src/lib/molecules/Charts/HeatMapChartD3/HeatMapChartD3.tsx

```diff
@@ -0,0 +1,350 @@
+                return null;
+            }
+
+            ctx.fillStyle = '#fff';
+            ctx.fillRect(
+                scaleX - BORDER_WIDTH,
+                scaleY - BORDER_WIDTH,
+                xScale.bandwidth() + BORDER_WIDTH * 2,
+                yScale.bandwidth() + BORDER_WIDTH * 2
+            );
+
+            const color = colorScale(value);
+            ctx.fillStyle = colorScale(value);
+            colorsSet.current.add(color);
+            ctx.fillRect(scaleX, scaleY, xScale.bandwidth(), yScale.bandwidth());
+        });
+    }, [canvasRef.current, uniqueData, xScale, yScale, colorScale, containerWidth]);
+
+    const xLabels = useMemo(
+        () =>
+            xGroups.map((name, i) => (
+                <span className="label" key={i}>
+                    {xAxisCategories[name]}
+                </span>
+            )),
+        [xGroups]
+    );
+
+    const yLabels = useMemo(
+        () =>
+            yGroups.reverse().map((name, i) => (
+                <span className="label" key={i}>
+                    {yAxisCategories[name]}
+                </span>
+            )),
+        [yGroups]
+    );
+
+    useEffect(() => {
+        const resizeHandler = () => setContainerWidth(containerRef.current?.getBoundingClientRect().width ?? 0);
+        window.addEventListener('resize', resizeHandler);
+        resizeHandler();
+
+        return () => window.removeEventListener('resize', resizeHandler);
+    }, [containerRef.current, legendLayout, enabledLegend]);
+
+    const closeTooltip = () => setHoveredCell(undefined);
+
+    const onMouseMoveOverCanvas = (e: MouseEvent<HTMLCanvasElement>) => {
+        const { offsetX, offsetY } = e.nativeEvent;
+
+        const currentIndex = uniqueData.findIndex(([x, y]) => {
+            const scaleX = xScale(x.toString());
+            const scaleY = yScale(y.toString());
+            if (!scaleX || !scaleY) {
+                return false;
+            }
+
+            return (
+                offsetX >= scaleX &&
+                offsetX < scaleX + xScale.bandwidth() &&
+                offsetY >= scaleY &&
+                offsetY < scaleY + yScale.bandwidth()
+            );
+        });
+
+        if (currentIndex > -1) {
+            const [x, y, value] = uniqueData[currentIndex];
+            const scaleX = xScale(x.toString());
+            const scaleY = yScale(y.toString());
+
+            if (scaleX && scaleY) {
+                setHoveredCell({
+                    x: scaleX + CONTAINER_PADDING + LEFT_SPACE + xScale.bandwidth() / 2,
+                    y: scaleY + TOP_SPACE,
+                    text: tooltipFormatter(String(value), xAxisCategories[x], yAxisCategories[y]),
+                    value
+                });
+            }
+        }
+    };
+
+    return (
+        <div className="chart-overflow-holder whiteDrillDown chart">
+            <BusyLoader isBusy={isLoading} className="proxy-content">
+                {!data?.length ? (
```

Comment:
The data has default value [] so you don't need to check.
```suggestion
                {!data.length ? (
```

Reply:
Correct, thanks.
Fixed.